File: iOS_Learning/AFNewtWorkingLearning/AFNewtWorkingLearning/ViewController.m
//
//  ViewController.m
//  AFNewtWorkingLearning
//
//  Created by 李冬阳 on 2017/5/24.
//  Copyright © 2017年 lidongyang. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view, typically from a nib.
  
//  NSURLSession
  
  /**
  NSLog(@"%@", [UIDevice currentDevice].systemVersion);
  
  NSURL *url = [NSURL URLWithString:@"http://120.25.226.186:32812/login"];
  NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url];
  request.HTTPMethod = @"POST";
  NSString *username = @"username";
  NSString *pwd = @"password";
  NSString *param = [NSString stringWithFormat:@"username=%@&pwd=%@", username, pwd];
  request.HTTPBody = [param dataUsingEncoding:NSUTF8StringEncoding];
  [request setValue:[UIDevice currentDevice].systemVersion forHTTPHeaderField:@"User-Agent"];
  [request setTimeoutInterval:10];
  
  NSURLResponse *response = nil;
  NSError *error = nil;
  NSData *data = [NSURLConnection sendSynchronousRequest:request returningResponse:&response error:&error];
  
  */

}


- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}


@end

File: iOS_Learning/iOS_Tools/iOS_Tools/iOS_Tools/ViewController.swift
//
//  ViewController.swift
//  iOS_Tools
//
//  Created by 李冬阳 on 2017/12/14.
//  Copyright © 2017年 lidongyang. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view, typically from a nib.
    
    /**
     Custom Notification  addObserve/ post 
     NotificationCenter.add(observe: self, selector: #selector(notification), name: .userLogin)
     
     Thread.sleep(forTimeInterval: 10)
     
     NotificationCenter.post(customNotification: .userLogin)
     */
    
    
    
  }

  @objc func notification() {
    print("notification is post success")
  }
  
  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }

  
  /**
   工具类
   
   1. 通知管理
   2. 颜色管理
   3. 字体管理
   4. log输出 (事件)
   5. 动画管理
   6. 异常处理
   7. 数据库管理
   8. 缓存处理
   9. 更换主题
   10. 线程管理
   */
  

}

